atlascope/core/rest/endpoints/detected_nucleus_endpoints.py

```diff
@@ -18,3 +25,35 @@ def get_queryset(self):
                 detection_dataset__id=detection_dataset_id,
             )
         return DetectedNucleus.objects.all()
+
+    @action(detail=True, methods=['get'])
+    def similar(self, request, pk=None):
+        nucleus = self.get_object()
+        queryset = similar_nuclei(nucleus)
+        page = self.paginate_queryset(queryset)
+        if page is not None:
+            serializer = SimilarNucleusSerializer(page, many=True)
+            return self.get_paginated_response(serializer.data)
+        serializer = SimilarNucleusSerializer(queryset, many=True)
+        return Response(serializer.data)
```

Comment:
does this mean the response may or may not be paginated? Does this change the shape of the response (e.g . keys "next" and "previous")?

Reply:
This is taken directly from DRF's [`ListModelMixin`](https://github.com/encode/django-rest-framework/blob/71e6c30034a1dd35a39ca74f86c371713e762c79/rest_framework/mixins.py#L33-L46). Ideally it should act as a paginated response in the same way other `list`-based DRF views would.